### DISCUSSION TITLE
Use .map and .map_dataframe with Seaborn object interface

### DISCUSSION THREAD

**Discussion by leodarrigade** [mwaskom/seaborn#3655] [Q&A] [answered]
Hi,

I am cross-posting here from Stackoverflow as I did not get any answer there.

I would like to apply the methods `seaborn.FacetGrid.map` and `seaborn.FacetGrid.map_dataframe` to a facet plot I build using the object interface. I cannot find a way to do it, any help would be appreciated.

For example, this is my figure

```python
import numpy as np
import pandas as pd

import matplotlib.pyplot as plt
import seaborn as sns   
import seaborn.objects as so

df = pd.DataFrame(np.arange(5*3).reshape(5,3), columns=["A","B","C"])
df["C"] = df["C"].apply(lambda x: x > 5)

fig = plt.figure()
fg = ((so.Plot(df, x="A", y= "B", group = "C")
       .add(so.Line(linewidth = 0.9)))
       .facet(col = "C")
       .on(fig)
       .plot())
```

And this is the function I would like to apply on each axe of the FacetGrid. This function write a text on each axe displaying the number of data used for the plot.

```python
def show_counts(color, data):
    ax = plt.gca()
    ax.text(0.5, 0.95, f'Count: {len(data)}', ha='center', transform=ax.transAxes)

fg.map_dataframe(show_counts)
```

However, from the last line of code, I get the following error :

```python
AttributeError: 'Plotter' object has no attribute 'map_dataframe'
```

Would there be a way to access the data of each axe and apply a function to it ?

**Comment by mwaskom** [comment, marked as answer]
Hi, unfortunately this is not possible, the objects interface does not use `FacetGrid` internally; they're completely separate systems.

**Reply by leodarrigade** [reply]
Ok thanks
